internal/ignition/ignition.go

```diff
@@ -1193,13 +1193,55 @@ func parseIgnitionFile(path string) (*config_latest_types.Config, error) {
 	return ParseToLatest(configBytes)
 }
 
-func (g *installerGenerator) getInterfaceIP(cidr string) string {
-	ip, _, err := net.ParseCIDR(cidr)
-	if err != nil {
-		g.log.Warnf("Failed to parse cidr %s for filling BMH CR", cidr)
-		return ""
+func (g *installerGenerator) selectInterfaceIPInsideMachineCIDR(interfaceCIDRs []string) string {
```

Comment:
Try to use GetPrimaryMachineCIDRIP, it should find you ip that matched cidr

Reply:
I can't use that function directly because it returns a single IP for the machine, and this code is trying to find the IP for a particular network interface and IP version.